src/test/ui/borrowck/slice-index-bounds-check-invalidation.rs

```diff
@@ -0,0 +1,63 @@
+// Test that we error if a slice is modified after it has been bounds checked
+// and before we actually index it.
+
+fn modify_before_assert_slice_slice(x: &[&[i32]]) -> i32 {
+    let mut x = x;
+    let z = [1, 2, 3];
+    let y = &[&z[..]];
```

Comment:
I think some explicit types on the variables in these functions would help the reader understand and focus on the relevant differences between them. Right now I have to sort of infer it from the fn name.